pkg/skaffold/graph/dependencies.go

```diff
@@ -33,14 +33,19 @@ import (
 // for testing
 var getDependenciesFunc = sourceDependenciesForArtifact
 
-// TransitiveSourceDependenciesCache provides an interface to evaluate and cache the source dependencies for artifacts.
-// This additionally includes the source dependencies from all other artifacts that are in the transitive closure of its artifact dependencies.
-type TransitiveSourceDependenciesCache interface {
-	ResolveForArtifact(ctx context.Context, a *latest_v1.Artifact) ([]string, error)
+// SourceDependenciesCache provides an interface to evaluate and cache the source dependencies for artifacts.
+type SourceDependenciesCache interface {
+	// TransitiveArtifactDependencies returns the source dependencies for the target artifact, including the source dependencies from all other artifacts that are in the transitive closure of its artifact dependencies.
+	// The result (even if an error) is cached so that the function is evaluated only once for every artifact. The cache is reset before the start of the next devloop.
+	TransitiveArtifactDependencies(ctx context.Context, a *latest_v1.Artifact) ([]string, error)
+	// SingleArtifactDependencies returns the source dependencies for only the target artifact.
+	// The result (even if an error) is cached so that the function is evaluated only once for every artifact. The cache is reset before the start of the next devloop.
+	SingleArtifactDependencies(ctx context.Context, a *latest_v1.Artifact) ([]string, error)
```

Comment:
Shd we be using SingleArtifactDependencies even for
1) Docker build context,
2) Kaniko build context
and else where in jib, custom?

Reply:
Transitive dependencies are only required for sync? or even in case of building.

Reply:
I checked all usages and updated them to use the Transitive dependencies only iff required.